### PR TITLE
PLU-115: [SST-6] frontend single step testing

### PR DESCRIPTION
**Pull request by pregnantboy**
### TL;DR

Implemented single-step execution and improved variable handling in the test environment.

### What changed?

- Replaced `EXECUTE_FLOW` mutation with `EXECUTE_STEP` on front end for single step testing

### How to test?

1. Create a new pipe with form trigger
2. Test for mock data
3. Create another step with any action (for ease of testing try calculator)
4. Test that new action step and verify that the mock data didnt change for the trigger step.
5. Now create an empty step 3 and step 4.
6. Set up step 4 with variables from prior steps. Click test step. It should work despite having step 3 not tested.
7. Now click test step on the formsg step and see that the mock data has changed.
8. Submit a form and see that it should reflect the new form submission as well

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#724** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/724?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#708** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/708?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#707** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/707?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#706** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/706?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#680** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/680?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#679** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/679?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#678** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/678?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/src/components/TestSubstep/index.tsx [hunk 1]
```diff
@@ -121,15 +121,19 @@ function TestSubstep(props: TestSubstepProps): JSX.Element {
...
+            stepId: step.id,
+          },
         },
-      },
-    })
-  }, [executeFlow, step.id])
+      })
+    } catch (e) {
+      console.error(e)
+    }
```
Comment (m0nggh) [inline packages/frontend/src/components/TestSubstep/index.tsx:135]: can i check when does this occur ah, is it more of a sanity check

**Review by m0nggh — APPROVED** [review]
tested old and new pipes flow
- test execution id gets updated and all the execution steps as well

new pipe
- individual steps can be tested
- old execution steps are deleted and replaced
- errored steps wont show up on the current step being tested
- if-then individual steps can be tested
- can just test an action without a trigger

lgtm, just wanted to point out we should remove the step position for `StepError` now?

**Comment by pregnantboy** [comment]
> lgtm, just wanted to point out we should remove the step position for `StepError` now?

After it's more stable and i can remove the rollback

File: packages/frontend/src/components/TestSubstep/index.tsx (hunk 1 above)
Comment (pregnantboy) [reply packages/frontend/src/components/TestSubstep/index.tsx:135]: yeap

**Comment by pregnantboy** [comment]
### Merge activity

* **Sep 4, 9:04 AM EDT**: @pregnantboy started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/708).
* **Sep 4, 9:05 AM EDT**: @pregnantboy merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/708).
